src/main/java/smile/identity/core/Signature.java

```diff
@@ -5,65 +5,120 @@
 import javax.crypto.Cipher;
 import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.Mac;
 import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.SecretKeySpec;
+
 import java.io.IOException;
+import java.nio.charset.StandardCharsets;
 import java.security.*;
 import java.security.spec.X509EncodedKeySpec;
 import java.util.Base64;
 
 public class Signature {
+
+    private static final String SHA_256 = "SHA-256";
+    private static final String HMAC_SHA_256 = "HmacSHA256";
+    public static final String SEC_KEY = "sec_key";
+    public static final String TIME_STAMP_KEY = "timestamp";
+    public static final String SIGNATURE_KEY = "signature";
+	
     private Integer partnerId;
-    private String api_key;
+    private String apiKey;
 
-    public Signature(String partner_id, String api_key) {
+    public Signature(String partner_id, String apiKey) {
         int partnerId = Integer.parseInt(partner_id);
         this.partnerId = partnerId;
-        this.api_key = api_key;
+        this.apiKey = apiKey;
+    }
+    
+	public JSONObject generateSignature() throws Exception {
+    	return generateSignature(System.currentTimeMillis());
+    }
+    
+	/***
+     *  Will generate a signature for the provided timestamp
+     * @param timestamp the timestamp to generate the signature from
+     * @return the JSON Object containing both the signature and related timestamp
+	 * @throws NoSuchAlgorithmException 
+	 * @throws InvalidKeyException
+     */
+    @SuppressWarnings({ "unchecked" })
+	public JSONObject generateSignature(Long timestamp) throws NoSuchAlgorithmException, InvalidKeyException {
+        JSONObject signatureObj = new JSONObject();
+    	signatureObj.put(TIME_STAMP_KEY, timestamp);
+    	
+    	SecretKeySpec secretKeySpec = new SecretKeySpec(apiKey.getBytes(), HMAC_SHA_256);
+    	
+        Mac mac = Mac.getInstance(HMAC_SHA_256);
+        mac.init(secretKeySpec);
+		mac.update(timestamp.toString().getBytes(StandardCharsets.UTF_8));
+		mac.update(partnerId.toString().getBytes(StandardCharsets.UTF_8));
+		mac.update("sid_request".getBytes(StandardCharsets.UTF_8));
+		
+		String signature = Base64.getEncoder().encodeToString(mac.doFinal());
+		signatureObj.put(SIGNATURE_KEY, signature);
+		
+		return signatureObj;
+    }
+    
+	/***
+     *  Will confirm the signature against a newly generated signature based on the same timestamp
+     * @param timestamp the timestamp to generate the signature from
+     * @param signature a previously signature, to be confirmed
+     * @return TRUE or FALSE
+     */
+    public Boolean confirmSignature(Long timestamp, String signature) {
+	    try {
+			return signature.equalsIgnoreCase((String) generateSignature(timestamp).get(SIGNATURE_KEY));
+		} catch (NoSuchAlgorithmException|InvalidKeyException e) {
+			return false;
+		}
     }
 
-    public String generate_sec_key(Long timestamp) throws Exception {
+    @SuppressWarnings("unchecked")
```

Comment:
Am curious to know why you're suppressing warnings here?

Reply:
Just convenience sake... `org.json.simple.JSONObject` is type-agnostic and the `put()` takes any of Java types without distinction. There's therefore no reason to keep this warning.